dirs/dirs.go

```diff
@@ -143,7 +156,18 @@ func SetRootDir(rootdir string) {
 	LocaleDir = filepath.Join(rootdir, "/usr/share/locale")
 	ClassicDir = filepath.Join(rootdir, "/writable/classic")
 
-	LibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
+	switch release.ReleaseInfo.ID {
+	case "fedora":
+		fallthrough
+	case "centos":
+		fallthrough
+	case "rhel":
+		DistroLibExecDir = filepath.Join(rootdir, "/usr/libexec/snapd")
+	default:
+		DistroLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
+	}
+
+	CoreLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
```

Comment:
@zyga Like I said, for now the core snap libexecdir thing is fine, but once it becomes possible for me to make a Fedora Snappy Core using a core snap based from the build artifacts from Fedora infrastructure, that will not be valid anymore.

If the change to support this difference is trivial, I'd prefer to see it handled now rather than later, since you're already messing with this code.